raft.c

```diff
@@ -1789,7 +1692,11 @@ static void handleRedisCommand(RedisRaftCtx *rr,RaftReq *req)
     /* If command is read only we don't push it to the log, but queue it
      * until we can confirm it's safe to execute (i.e. still a leader).
      */
-    if (checkReadOnlyCommandArray(&req->r.redis.cmds)) {
+    unsigned int cmd_flags = CommandSpecGetAggregateFlags(&req->r.redis.cmds);
+    if (cmd_flags & CMD_SPEC_UNSUPPORTED) {
+        RedisModule_ReplyWithError(req->ctx, "ERR not supported by RedisRaft");
+        goto exit;
+    } else if (cmd_flags & CMD_SPEC_READONLY) {
```

Comment:
I dont understand how this works.  it seems that if in a multi one has multiple commands and one if CMD_SPEC_READONLY, the flag would be set, even if there were other write commands mixed in?

Reply:
Yes, that needs to be fixed. The reason it worked was that when processing commands from a user, there will always be just a single command. The only case where we currently have multiple commands batched in a single entry is when processing a log entry that is a result of `MULTI/EXEC`.

Reply:
@sjpotter The statement above is wrong, this *does* affect the leader as well. Fixing and adding a test.